src/Illuminate/Contracts/Database/QueryBuilder.php

```diff
@@ -0,0 +1,1195 @@
+<?php
+
+namespace Illuminate\Contracts\Database;
+
+use Closure;
+
+interface QueryBuilder
```

Comment:
Can we name this `Illuminate\Contracts\Database\Query\Builder` instead?

Reply:
I thought about this, and I'm fine either way. I think having three different `Builder` types will be a bit trickier, since you may also need to import one or both other implementations into the same file, forcing you to import `as BuilderContract` and `as BaseBuilder` or something like that, but I'm fine with renaming it for consistency's sake.

Reply:
I think it's best to keep this consistent with the rest for of the framework. Everywhere else we where we also use the same contract interface names as the implementations it's also always `BuilderContract`. And since the purpose of this PR is to avoid ambiguity I'd say the contract reference will become the standard while the concrete implementations will become the exception. And if they're needed, they're more clearly referenced as `EloquentBuilder` and `QueryBuilder`.